### PR TITLE
OCPBUGS-6634: Enable building tectonic-console-builder on arm64 and fix the Dockerfile.product to work with no cached artifacts

### PR DESCRIPTION
**Pull request by aleskandro**
This PR will allow:
1. building the console-tecotonic-builder (`Dockerfile.builder`) and the console (`Dockerfile`) on architectures different than amd64;
2. building the console using the `Dockerfile.product` outside the internal build engines.

I disabled the installation of google-chrome in the `Dockerfile.builder` as it is not shipped for arm64: chromium is available, but I'm not sure it is ok for the purpose it has in the tests.

Still, `console-tectonic-builder` will only allow building the console on arm64 once nodeJs is bumped to v14.17.0 (see nodejs/help#3202). As the idea is to switch using a unique Dockerfile for both the CI and the shipped product, this PR will not address the version bump, and the builds for arm64 can work from the `Dockerfile.product`. 

After this is merged, https://github.com/openshift/release/pull/35527 can land, and we can start using `Dockerfile.product` in the CI and for OKD.

The changes to the `Dockerfile.product` have been tested in prow through a temporary commit. log [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/12444/pull-ci-openshift-console-master-images/1615973329284370432/artifacts/build-logs/console.log).

cc @jeffdyoung @LorbusChris

### CODE REVIEW

**Comment by jhadvig** [comment]
Thanks for the change @aleskandro 👍  Please let me know once you change the image in the `.ci-operator.yaml` file

**Comment by aleskandro** [comment]
Based on the discussion in https://redhat-internal.slack.com/archives/C6A3NV5J9/p1674051149090609, I'm giving a shot to make the `Dockerfile.product` compatible for both ART and Prow builds.

**Comment by aleskandro** [comment]
The changes to the Dockerfile.product have been tested in prow through a temporary commit. log [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/12444/pull-ci-openshift-console-master-images/1615973329284370432/artifacts/build-logs/console.log).

They will also be ok to build on arm64 as already done by ART.

Though, I would also check-in the ones in the Dockerfile.builder to make it general enough to work on other architectures

/cc @spadgett

File: Dockerfile.builder [hunk 1]
```diff
@@ -55,7 +56,8 @@ RUN cd /tmp && \
     ln -s /usr/local/yarn/bin/yarn /usr/local/bin/yarn
 
 # Install Chrome for installer gui tests
-RUN wget --quiet -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
+RUN test "$(go env GOARCH)" = amd64 || exit 0; \
```
Comment (aleskandro) [inline Dockerfile.builder:59]: Can the test suite use chromium and provide reliable test results? We could install it at least for non-amd64 images.

**Comment by spadgett** [comment]
Thanks @aleskandro. I think it would be good to have ART review the Dockerfile.product changes.

/cc @sosiouxme

**Comment by LorbusChris** [comment]
@aleskandro if `Dockerfile` is going to go away, why not replace it with a symlink to `Dockerfile.product` here in this PR?

**Comment by aleskandro** [comment]
> @aleskandro if `Dockerfile` is going to go away, why not replace it with a symlink to `Dockerfile.product` here in this PR?

We could also rename Dockerfile.product to Dockerfile then and change the ref in https://github.com/openshift/ocp-build-data/blob/openshift-4.13/images/openshift-enterprise-console.yml#L3

**Comment by LorbusChris** [comment]
Sure, it's just when the two are symlinks the config changes in `release` become optional/merely cleanup.

**Review by LorbusChris — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/approve

**Comment by aleskandro** [comment]
The failing tests seem unrelated to me, can we skip?

**Comment by jhadvig** [comment]
```
1) Testing uninstall of Business Automation Operator
       "before each" hook for "attempts to uninstall the Operator and delete all Operand Instances, shows 'Error Deleting Operands' alert":
     CypressError: Timed out retrying after 30050ms: `cy.click()` failed because this element:
`<a data-test="nav" class="pf-c-nav__link pf-m-current" href="/k8s/ns/test-omyaf/operators.coreos.com~v1alpha1~ClusterServiceVersion" aria-current="page">Install...</a>`
is being covered by another element:
`<div class="ReactModal__Overlay ReactModal__Overlay--after-open co-overlay">...</div>`
```

**Pull request merged**